src/queries/queries.jobs.repository.ts

```diff
@@ -0,0 +1,116 @@
+import { Effect as E, pipe } from "effect";
+import {
+  type InsertOneResult,
+  type StrictFilter,
+  type StrictUpdateFilter,
+  UUID,
+} from "mongodb";
+import {
+  DatabaseError,
+  DocumentNotFoundError,
+  type PrimaryCollectionNotFoundError,
+} from "#/common/errors";
+import { CollectionName, checkPrimaryCollectionExists } from "#/common/mongo";
+import type { AppBindings } from "#/env";
+import type { QueryJobDocument } from "./queries.types";
+
+export function toQueryJobDocument(queryId: UUID): QueryJobDocument {
+  const now = new Date();
+
+  return {
+    _id: new UUID(),
+    _created: now,
+    _updated: now,
+    status: "pending",
+    queryId,
+  };
+}
+
+export function insert(
+  ctx: AppBindings,
+  document: QueryJobDocument,
+): E.Effect<
+  InsertOneResult<QueryJobDocument>,
+  PrimaryCollectionNotFoundError | DatabaseError
+> {
+  return pipe(
+    checkPrimaryCollectionExists<QueryJobDocument>(
+      ctx,
+      CollectionName.JobsQueries,
+    ),
+    E.flatMap((collection) =>
+      E.tryPromise({
```

Comment:
I haven't used it yet, but I think there's a `E.tryMapPromise` which will simplify this pattern